File: Python/Algorithms/728.py
class Solution:

    def selfDividingNumbers(self, left, right):
        ret = []
        divmod_ = divmod
        for num in range(left, right+1):
            # Find number of digits.
            digits, temp = 0, num
            while temp:
                digits, temp = digits + 1, temp // 10
            div = 10 ** (digits - 1)
            # go through digits and count number of
            # digits that divide num.
            count, red = 0, num
            while div:
                d, red = divmod_(red, div)
                if d and not num % d:
                    count = count + 1
                else:
                    break
                div = div // 10
            # If all digits divide add it to result.
            if count == digits:
                ret.append(num)

        return ret

File: Rust/Algorithms/728.rs
impl Solution {
    pub fn self_dividing_numbers(left: i32, right: i32) -> Vec<i32> {
        let mut res = Vec::new();
        for num in left..right+1 {
            // Count number of digits in num
            let (mut digits, mut temp) = (0, num);
            while temp != 0 {
                digits += 1;
                temp /= 10;
            }
            let mut div = (10_u32).pow(digits - 1) as i32;
            // Go through digits and count how many 
            // divide num
            let (mut count, mut red) = (0, num);
            while div != 0 {
                let d = red / div;
                red %= div;
                if d != 0 && num % d == 0 {
                    count += 1;
                } else {
                    break;
                }
                div /= 10;
            }
            // Equality means we've counted all digits.
            // Add it.
            if count == digits {
                res.push(num);
            }
        }
        
        res
    }
}
